solutions/usaco-834.mdx

```diff
@@ -62,4 +62,25 @@ public class OutOfSorts {
 }
 ```
 </JavaSection>
+
+<PySection>
+
+```python
+with open('sort.in') as read:
+	n, fix = int(read.readline()), 111111
+	l, locations = [0] * n, {}
+
+	for i in range(n):
+		idx = int(read.readline()) * fix + i
+		l[i], locations[idx] = idx, i
```

Comment:
this would be easier to understand if you picked a more informative name than `idx`. it's also a little confusing how you use `i` as an index here but then as an array value on line 80

Reply:
is val better? since it's a value in an array (unsorted)

Reply:
sure